src/server/system_services/system_store.js

```diff
@@ -331,7 +331,12 @@ class SystemStoreData {
         _.each(COLLECTIONS, col => {
             _.each(col.mem_indexes, index => {
                 _.each(this[col.name], item => {
-                    let key = _.get(item, index.key || '_id').valueOf();
+                    const field = _.get(item, index.key || '_id');
+                    if (!field) {
+                        dbg.warn(`SystemStoreData: Item ${col.name}[${item._id}] could not be indexed for ${index.name}
+                            (Index field ${index.key} not found on item), skipping`);
+                    }
+                    const key = field.valueOf();
```

Comment:
means we'd add map[undefined] ? do we want it ?

Reply:
Added missing return